demos/draggable-line-demo.js

```diff
@@ -0,0 +1,31 @@
+let w; //graph window
+let myLine;
+let draggable;
+
+function setup() {
+  createCanvas(400, 400);
+
+  const xOrigin = width / 2;
+  const yOrigin = height / 2;
+  const xScale = 10; //px per unit
+  const yScale = 10; //px per unit
+  w = createGraphWindow(xOrigin, yOrigin, xScale, yScale);
+
+  myLine = w.createLine(-1, -2, 3, 4);
+  myLine.setIsDraggable(true)
```

Comment:
Let's come to a tentative consensus on the 'setIsDraggable' name before we merge this, since it affects the interface. Here are some issues to consider.

**Comparison to our naming scheme for event listeners:** The name is consistent with the naming scheme for event listener setters, such as setObjectIsHovered(). However, "setIsDraggable" sets a Boolean, whereas "setObjectIsHovered" sets a listener that returns a Boolean. This may be okay, since there are two cues that these setters accept different kinds of input. The biggest cue is the "-able" and "-ed" endings. The other cue is that the passive "isHovered" is paired with "Object" (the thing being hovered). So, _the "setIsDraggable" name seems okay from this perspective._

**Comparison to p5**
Some cases where p5 uses "set" are as follows:

- setAttributes() is a general setter for WebGL that can be called like setAttributes(key, value) or setAttributes(obj)
- setMoveThreshold()
- setShakeThreshold() 
- set() for setting pixels or writing a whole image to the display window
- setCamera()
- .set() for p5 vectors and for p5.Camera objects (p5.Camera objects also have .setPosition())

Some cases where p5 uses "is" in a function name: isLooping() and keyIsDown(). Neither is a setter. Both return Booleans, as we'd expect based on the usual convention.

Since the use of "setIsDraggable" seems consistent with usage in p5, and the use of "isDraggable" as a setter seems inconsistent with p5, the name "setIsDraggable" seems preferable. _From this perspective, the current name seems good._

**Comparison with the combined getter/setter approach of Konva.js:** In this approach, methods like .stroke() do two things: they set the relevant property based on the arguments that are supplied, and they also return the value of the property. So, for example, calling .stroke() without arguments just gets the object's current stroke. In the same way, Konva.js has a .draggable() method that acts as a getter and setter for a Boolean "draggable" flag. However, "isDraggable" makes it more clear that we're dealing with a Boolean (due to the naming convention for Booleans), and .isDraggable() probably shouldn't be used as a setter, to avoid conflict with p5. _Again, "setIsDraggable" seems best._

*Tentative conclusion:* The current name .setIsDraggable() seems to work best. For other functions like .stroke(), we can have a separate discussion to break down the pros and cons of combined get/set functionality. As with many things related to getters and setters, there's debate about whether combined getters/setters are generally helpful or unhelpful. Some pros and cons are discussed here:

- [What are the benefits of combined getter and setter methods?](https://stackoverflow.com/questions/17651297/what-are-the-benefits-of-combined-getter-and-setter-methods)
- [Independent getter/setter methods, or combined?](https://stackoverflow.com/questions/6674784/independent-getter-setter-methods-or-combined)

Reply:
I think what's missing from this analysis is that setIsDrabble, to me, suggests that there is a boolean inside that we are setting.  Other developers familiar with common naming conventions will also think that.  It's true that you can look at the signature or docs and see that it's a callback, but that's elevating the importance of documentation, relative to making the API intuitive.

Reply:
Personally, I didn't pick up on the cues, I don't think.  The naming convention is too ingrained for me.

Reply:
I don't really see the "-ed" or "-abble" as cues the way you do, and at least in the p5 editor, there isn't going to be any autocomplete suggestion that it takes a callback.